From 6893e740fe5e50b11e415fae969b37596d6df571 Mon Sep 17 00:00:00 2001
From: Louis Marelle <louis.marelle@latmos.ipsl.fr>
Date: Mon, 22 Oct 2018 12:16:59 +0200
Subject: [PATCH 12/70] Add aerosol sedimentation aloft in MOSAIC

Add a new module in chem/, module_mosaic_settling, calculating sedimentation
above the lowest model level for MOSAIC aerosols. The released model only
includes aerosol sedimentation at the surface, and its contribution to dry
deposition. This module is called from dry_dep_driver in chem/

The new module calculates settling velocities with the same code than
the one already used in MOSAIC for sedimentation at the surface, and
then performs sedimentation (bringing aerosols from levels k+1 to k
etc.) using an explicit method. It performs sedimentation for aerosol
mass, aerosol number (calculated from mass/volume based on the bin
center diameter) and aerosol water.

The module could be optimized for speed by rewriting the code structure
to remove one vertical k loop, by only performing sedimentation for
aerosols above a certain particle diameter (1 um?), and by not
performing sedimentation when aerosol mass mixing ratios are below some
minimal value.
---
 Registry/registry.chem | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Registry/registry.chem b/Registry/registry.chem
index 9e99a48..725c9d1 100644
--- a/Registry/registry.chem
+++ b/Registry/registry.chem
@@ -3866,6 +3866,8 @@ rconfig   integer     chem_in_opt         namelist,chem          max_domains
 rconfig   integer     phot_opt            namelist,chem          max_domains    0       rh    "phot_opt"            ""      ""
 rconfig   integer     gas_drydep_opt      namelist,chem          max_domains    1       rh    "gas_drydep_opt"      ""      ""
 rconfig   integer     aer_drydep_opt      namelist,chem          max_domains    1       rh    "aer_drydep_opt"      ""      ""
+#LMarelle add an option for gravitational settling in MOSAIC
+rconfig   integer     mosaic_aer_settling_opt    namelist,chem          max_domains    0       rh    "mosaic_aer_settling_opt"    ""      ""
 rconfig   integer     diagnostic_chem     namelist,chem          max_domains    1       rh    "diagnostic_chem"     ""      ""
 rconfig   integer     aero_diag_opt       namelist,chem          max_domains    0       rh    "diagnostic_aerosol"  ""      ""
 rconfig   integer     aero_cw_diag_opt    namelist,chem          max_domains    0       rh    "diagnostic_cw_aerosol"     ""      ""
-- 
2.31.1

